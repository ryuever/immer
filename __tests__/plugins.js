import produce, {
	setUseProxies,
	produceWithPatches,
	applyPatches
} from "../src/immer"

describe("ES5 plugins should throw if no proxies are available", () => {
	beforeEach(() => {
		setUseProxies(false)
	})

	afterEach(() => {
		setUseProxies(true)
	})

	test("error when using ES5", () => {
		expect(() => {
			produce({}, function() {})
		}).toThrowErrorMatchingSnapshot()
	})
})

test("error when using Maps", () => {
	expect(() => {
		produce(new Map(), function() {})
	}).toThrowErrorMatchingSnapshot()
})

test("error when using patches - 1", () => {
	expect(() => {
		produce(
			{},
			function() {},
			function() {}
		)
	}).toThrowErrorMatchingSnapshot()
})

test("error when using patches - 2", () => {
	expect(() => {
		produceWithPatches({}, function() {})
	}).toThrowErrorMatchingSnapshot()
})

test("error when using patches - 3", () => {
	expect(() => {
		applyPatches({}, [])
	}).toThrowErrorMatchingSnapshot()
})
