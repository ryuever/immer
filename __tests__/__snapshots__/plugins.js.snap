// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ES5 plugins should throw if no proxies are available error when using ES5 1`] = `"[Immer] The plugin for 'ES5' has not been loaded into Immer. To enable the plugin, import and call \`enableES5()\` when initializing your application."`;

exports[`error when using Maps 1`] = `"[Immer] The plugin for 'MapSet' has not been loaded into Immer. To enable the plugin, import and call \`enableMapSet()\` when initializing your application."`;

exports[`error when using patches - 1 1`] = `"[Immer] The plugin for 'Patches' has not been loaded into Immer. To enable the plugin, import and call \`enablePatches()\` when initializing your application."`;

exports[`error when using patches - 2 1`] = `"[Immer] The plugin for 'Patches' has not been loaded into Immer. To enable the plugin, import and call \`enablePatches()\` when initializing your application."`;

exports[`error when using patches - 3 1`] = `"[Immer] The plugin for 'Patches' has not been loaded into Immer. To enable the plugin, import and call \`enablePatches()\` when initializing your application."`;
